graph.py

```diff
@@ -55,12 +59,41 @@ def has_node(self, value):
     def neighbors(self, value):
         """Return a list of all nodes connected to the node containing 'value' by edges; raise an error if 'value' is nto in graph."""
         if value in self._storage:
-            return [edge_value[1] for edge_value in self._storage[value]]
+            return [edge_value for edge_value in self._storage[value]]
         else:
             raise ValueError("Value not in graph.")
 
     def adjacent(self, value1, value2):
         """Return if there is an edge connecting 'value1' and 'value2'; raise an error if either of the supplied values are not in graph."""
         if value1 not in self._storage or value2 not in self._storage:
             raise ValueError("One of the values are not in the graph")
-        return (value1, value2) in self._storage[value1]
+        return value2 in self._storage[value1]
+
+    def depth_first_traversal(self, start_val):
+        """Return a list of depth_first_traversal starting at start_val"""
+        path = [start_val]
+        track = Stack()
+        for edge in self._storage[start_val]:
+            track.push(edge)
+        while track:
```

Comment:
`track` will always be truthy.  Instead of using your Stack, use a list here.  Use it in a way that the needed oeprations are O(1)